docs/06-contributors/999-rfcs/2023-04-16-website-resource.md

```diff
@@ -0,0 +1,111 @@
+Stage: Fill in with current stage in the RFC lifecycle
+Stage Date: Fill in with date of last stage change
+-->
+
+> Implementing the website resource following the [SDK spec](../../04-reference/wingsdk-spec.md#website).
+
+<!--
+This RFCs have 3 sections: Requirements -> Design -> Implementation. We intentionally start with *Design* since it
+is a great way to introduice the feature to readers.
+-->
+
+## Implementation
+
+### Implementation for aws
+
+Following a few AWS guides on the internet (brought in the end of the section), it seems that the common way to serve a static website is by using a bucket.
+
+After creating a public bucket, we'll upload the folder from the relative path.
+AWS requires a index document name (to serve on the "/" path).
+We'll start by hardcoding it to be "index.html", and can add it later to the website props upon request.
+
+The last thing will be a registration to the chosen domain (if exists).
+
+The website resource supports preflight code only.
+
+#### References:
+
+[Tutorial: Configuring a static website on Amazon S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html)
+[Use your domain for a static website in an Amazon S3 bucket](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/getting-started-s3.html)
+
+### Implementation for the simulator
+
+The website will be implemented by a static express server:
+
+```js
+const app = Express();
+
+app.use(express.static("<resolved path to the website files folder>"));
+
+app.listen(
+  "<port, ideally will be specified by the user, otherwise generated>"
+);
+```
+
+Since the website is served on the local machine, we cannot specify a domain, nor can we add a CDN while deploying to the simulator.
+
+### point for thought:
+
+I wonder if we should support .env files, in the future, (or any other kind of configuration-per-target system) for specifying different value for different targets:
+
+For example:
+
+In the wing file:
+
+```
+let main_website = new cloud.Website({path: "./website/main/dist", domain: MAIN_WEBSITE_DOMAIN})
+let back_office_website = new cloud.Website({path: "./website/back-office/dist", domain: BO_WEBSITE_DOMAIN})
+```
+
+In tf-aws.env/ prod.env:
+
+```
+MAIN_WEBSITE_DOMAIN= "my-company.com"
+BO_WEBSITE_DOMAIN= "bo.my-company.com"
+```
+
+In sim.env/ dev.env:
+
+```
+MAIN_WEBSITE_DOMAIN= "localhost:3000"
+BO_WEBSITE_DOMAIN= "localhost:4000"
+```
+
+This, of course, rises more questions then answers-
+how can we support different development environments? How do we tell wing which env file to use? Can we share those files between targets? (for example, running the same file for tf-gcp and tf-azure, and a different one for the sim).
+
+- Website props are brought here for reference:
+
+```ts
+struct WebsiteProps {
+  /**
+   * Local path to the website's static files, relative to the Wing source file.
+   * @example "./dist"
+   */
+  path: str;
```

Comment:
Thought: ideally the path should be relative to the source file (and not to the current working directory).

What I mean is that if I pass `./site` I actually want the sdk to receive the path `__dirname + "./site"`. Ideally this should be implicit, so we might need some language support for this.

@Chriscbr what do you think?

Reply:
I started coding this to see if any unexpected issues are popping up, and one of the first things I notice is that I can't use relative paths to the .w files yet, since the SDK is located in the node_modules (maybe even the global ones) and "has no idea" where is the .w located, I might need some help with that :)
(currently using an absolute path to bypass it, but it isn't nice...)

Reply:
I think Elad's thought is what I tried to include in the SDK spec 😅 sorry if the wording wasn't clear.

One idea for how to implement this could be to set a `WING_SOURCE_DIR` environment variable inside the `wing` CLI package [here](https://github.com/winglang/wing/blob/9db4d03c13f5689dca056599bdf368583235fc7e/apps/wing/src/commands/compile.ts#L97)-ish, and assign to it the absolute path to the directory, and then use the environment variable within the SDK code through `process.env` to create the full path name.

Reply:
Works perfectly, thanks @Chriscbr!!